src/Nancy/Json/Simple/NancySerializationStrategy.cs

```diff
@@ -0,0 +1,231 @@
+        }
+
+        /// <summary>
+        /// Register custom <see cref="JavaScriptPrimitiveConverter"/>
+        /// </summary>
+        /// <param name="javaScriptPrimitiveConverters">An array of <see cref="JavaScriptPrimitiveConverter"/></param>
+        public void RegisterConverters(IEnumerable<JavaScriptPrimitiveConverter> javaScriptPrimitiveConverters)
+        {
+            this.primitiveConverters.AddRange(javaScriptPrimitiveConverters);
+        }
+
+        /// <summary>
+        /// Formats a property name to a JSON field name
+        /// </summary>
+        /// <param name="clrPropertyName">The property name to format</param>
+        /// <returns></returns>
+        protected override string MapClrMemberNameToJsonFieldName(string clrPropertyName)
+        {
+            return this.retainCasing
+                ? base.MapClrMemberNameToJsonFieldName(clrPropertyName)
+                : clrPropertyName.ToCamelCase();
+        }
+
+        /// <summary>
+        /// Deserialize an object
+        /// </summary>
+        /// <param name="value">The object to deserialize</param>
+        /// <param name="type">The type of object to deserialize</param>
+        /// <returns></returns>
+        public override object DeserializeObject(object value, Type type)
+        {
+            if (type.IsEnum || (ReflectionUtils.IsNullableType(type) && Nullable.GetUnderlyingType(type).IsEnum))
+            {
+                var typeToParse = ReflectionUtils.IsNullableType(type) 
+                    ? Nullable.GetUnderlyingType(type) 
+                    : type;
+
+                return value == null
+                    ? null
+                    : Enum.Parse(typeToParse, value.ToString(), true);
+            }
+
+            var primitiveConverter = this.FindPrimitiveConverter(type);
+            if (primitiveConverter != null)
+            {
+                return primitiveConverter.Deserialize(value, type);
+            }
+
+            var valueDictionary = value as IDictionary<string, object>;
+            if (valueDictionary == null)
+            {
+                return base.DeserializeObject(value, type);
+            }
+
+            var javascriptConverter = this.FindJavaScriptConverter(type);
+            if (javascriptConverter != null)
+            {
+                return javascriptConverter.Deserialize(valueDictionary, type);
+            }
+
+            if (!type.IsGenericType)
+            {
+                return base.DeserializeObject(value, type);
+            }
+
+            var genericType = type.GetGenericTypeDefinition();
+            var genericTypeConverter = this.FindJavaScriptConverter(genericType);
+
+            if (genericTypeConverter == null)
+            {
+                return base.DeserializeObject(value, type);
+            }
+
+            var values = new Dictionary<string, object>(StringComparer.OrdinalIgnoreCase);
+            var genericArguments = type.GetGenericArguments();

```

Comment:
Needs to be type.GetTypeInfo().GetGenericArguments() or it won't work on the CoreCLR